ostree-upload-server.py

```diff
@@ -96,47 +122,84 @@ def upload_bundle():
-        upload = request.files['file']
-        if upload.filename == "":
-            return "no filename in upload\n"
-        (f, real_name) = tempfile.mkstemp(dir=app.config['UPLOAD_FOLDER'])
-        os.close(f)
-        upload.save(real_name)
-        task_list.add_task(Task(upload.filename, real_name))
-        return "task added\n"
-
+        print("/upload: POST request start")
+        with active_upload_counter:
+            if 'file' not in request.files:
+                return "no file in POST\n"
+            upload = request.files['file']
+            if upload.filename == "":
+                return "no filename in upload\n"
+            (f, real_name) = tempfile.mkstemp(dir=app.config['UPLOAD_FOLDER'])
+            os.close(f)
+            upload.save(real_name)
+            task_list.add_task(Task(upload.filename, real_name))
+            print("/upload: POST request completed for " + upload.filename)
+            return "task added\n"
+
+class Workers:
+    def __init__(self):
+        self.workers = []
+        self.quit_workers = Event()
+    def start(self, task_list, worker_func, worker_count=4):
+        for i in range(worker_count):
+            w = Greenlet.spawn(worker_func, task_list.get_queue(), self.quit_workers)
+            self.workers.append(w)
+    def stop(self):
+        self.quit_workers.set()
+        for w in self.workers:
+            w.join()
+        self.quit_workers.clear()
 
 if __name__=='__main__':
-    print("Starting server on %d..." % PORT)
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-w", "--workers", type=int, default=4,
+                        help="number of uploads to process in parallel")
+    parser.add_argument("-p", "--port", type=int, default=5000,
+                        help="HTTP server listen port")
+    args = parser.parse_args()
 
-    # TODO: Add argparse for these settings
-    worker_count = 4
-    workers = []
+    print("Starting server on %d..." % args.port)
 
-    quit_workers = Event()
-    for i in range(worker_count):
-        w = Greenlet.spawn(worker, task_list.get_queue(), quit_workers)
-        workers.append(w)
+    workers = Workers()
+    workers.start(task_list, worker, args.workers)
 
-    http_server = WSGIServer(('', PORT), app)
+    http_server = WSGIServer(('', args.port), app)
     http_server.start()
 
-    print("Server started on %s" % PORT)
+    print("Server started on %s" % args.port)
 
     # loop until interrupted
     while True:
         try:
             gsleep(5)
             task_list.join()
+            print("task queue empty, " + str(active_upload_counter.count) + " uploads ongoing")
+            time_since_maintenance = time() - latest_maintenance_complete
+            time_since_task = time() - latest_task_complete
+            print("{:.1f} since last task, {:.1f} since last maintenance".format(
+                        time_since_task,
+                        time_since_maintenance))
```

Comment:
I'll also add a debug flag for these kinds of messages - it's a bit too verbose at the moment